lib/puppet/util/autoload.rb

```diff
@@ -45,12 +45,11 @@ def mark_loaded(name, file)
       loaded[name] = [file, File.mtime(file)]
     end
 
-    def changed?(name)
+    def changed?(name, env = Puppet.lookup(:current_environment))
```

Comment:
Several autoloader instance and class methods like (this one) could be made private. Some of the autoloader unit tests would need to be updated (to stop testing private methods). Alternatively, we could mark them with `@api private`. Thoughts @pcarlisle @hlindberg?

Reply:
not having looked at the tests, either option seems fine there